docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,151 @@
+    * The [paused](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#pausing-reconciliation) annotation, is similar to the `detached` annotation we're already using - but, it's also pausing the reconcile loop. Thus, keeping the status annotation intact.
+
+For BMHs in the passive hub: (In assisted-service -> `bmh_agent_controller`)
+* When the BMH is applied, the status will be missing - since reconcile is paused.
+  * To restore the original status:
+    * Remove the `baremetalhost.metal3.io/paused` annotation
+      * Status is restored, but `baremetalhost.metal3.io/status` annotation is removed.
+    * Hence, add the status and paused annotation back
+      * By detecting a BMH with `provisioned` state and a missing status annotation.  
+
+#### Solution for deleted Agent CRs
+
+1. Don't delete the Agent CR when an associated host doesn't exits on DB.
+   * Remove relevant [code](https://github.com/openshift/assisted-service/blob/5bbf696fc0cc5a220fd8a0d1e408ab0bcc889e3f/internal/controller/controllers/agent_controller.go#L143-L151) from AgentController.
+
+2. Restore the Host record in DB according to properties from the Agent CR.
+   * Seems we can infer more than the mandatory properties:
+     * host.ID <-- agent.Name
+     * host.Role <-- agent.Role
+     * host.InfraEnvID <-- agent.metadata.annotations['infraenvs.agent-install.openshift.io']
+     * host.Kind <-- 'Host' or 'AddToExistingClusterHost' if cluster is already installed
+     * host.InstallerArgs <-- agent.InstallerArgs
+     * host.RequestedHostname <-- agent.Hostname
+   * Some of the inventory data can be restored from annotations.
+     E.g.:
+     * inventory.agent-install.openshift.io/cpu-architecture: x86_64
+     * inventory.agent-install.openshift.io/cpu-virtenabled: "true"
+     * inventory.agent-install.openshift.io/host-isvirtual: "true"
+     * inventory.agent-install.openshift.io/host-manufacturer: RedHat
+     * inventory.agent-install.openshift.io/host-productname: KVM
+     * inventory.agent-install.openshift.io/storage-hasnonrotationaldisk: "false"
+   * Note: We might need to add more annotations for additional inventory data.
+   
+### Risks and Mitigations
+
+* In general, these changes might affect flows on spoke cluster (before and after restore), need to ensure no regressions on all scenarios.
+* Adding the 'paused' annotation means a stale BMH status, need to check if it's crucial.
+* Keeping the Agent CR and restoring values into Host in DB could affect current flows (e.g. BindHost, UnbindHost, etc).
+* Restoring Hosts will result in partial records in DB. E.g. missing Inventory - could be challenging to tackle (I.e. probably need to add annotations to Agent).
+
+### Open Questions
+
+* Any specific concerns for the ConvergedFlow/Non-ConvergedFlow?
+* How seamlessly should we support restoring of 'in-progress' clusters (as the focus is on 'installed' clusters)?
```

Comment:
For this I think we just need a clear answer as to what will happen along with the ability to recover.

DR is a necessarily uncontrolled scenario and some things will be in an unstable state when it happens. As long as nothing gets "stuck" on the restore side we should be fine.

Also something to think about is that the most recent backup might not reflect the current state of the world.
Imagine a scenario where the backup has an unbound agent, but that agent had actually been installed since the backup was taken. Also the opposite might be true. Obviously there's not much we can do to make this "correct" in the restore, but it's worth thinking about.

Reply:
We should probably ask @racedo about the requirements here.  My opinion is that we don't need to continue with the installations that are in progress.  Ideally, the cluster and agent resources should either indicate that an error occurred or allow the user to retry.  I'm guessing that if we do no work here then the agents will show up as disconnected, the cluster will show up as waiting for the agents, and installation will start once agents reboot with the new ISO.  This is also acceptable IMO.